src/main/webapp/homepage_script.js

```diff
@@ -307,15 +308,35 @@ begins the processes of retrieving the user's location*/
 function getTasksForUserLocation() {
     const script = document.createElement("script");
     script.type = "text/javascript";
-    script.src =  "https://maps.googleapis.com/maps/api/js?key=" + MAPSKEY + "&callback=initialize&language=en";
+    script.src =  "https://maps.googleapis.com/maps/api/js?key=" + MAPSKEY + "&callback=initialize&libraries=places&language=en";
     script.defer = true;
     script.async = true;
     document.head.appendChild(script);
-	
+
     // Once the Maps API script has dynamically loaded it gets the user location,
     // waits until it gets an answer updates the global userLoaction variable and then calls
     // fetchTasks and displayTasks
 	window.initialize = function () {
+        let placeAutocomplete = new google.maps.places.Autocomplete(document.getElementById("place-input"));
+        placeAutocomplete.setFields(['geometry']);
+        google.maps.event.addListener(placeAutocomplete, 'place_changed', function() {
+                let place = placeAutocomplete.getPlace();
+                if (place.geometry != undefined) {
+                    userLocation = place.geometry.location.toJSON();
+                } else {
+                    userLocation = userActualLocation;
+                }
+                toNeighborhood(userLocation)
+                        .then(() => fetchTasks(currentCategory, "clear"))
+                        .then(response => {
+                                taskGroup = response;
+                                displayTasks(response);
+                            })
+                        .catch(() => {
+                            console.error("User location and/or neighborhood could not be retrieved");
```

Comment:
If the user location or neighborhood could not be retrieved, do we have to report an error in the console? I think maybe just provide the users with a proper method to select their own positions could be a better solution.

Reply:
Ah, you're right! I'll remove the console error message and update the message displayed to the user in the site to reflect the search box changes, thanks!